blockonomics-woocommerce.php

```diff
@@ -156,29 +156,62 @@ public function process_payment($order_id)
                 $cancel_url = add_query_arg('cancelled', true, $cancel_url);
                 $cancel_url = add_query_arg('order_key', $order->order_key, $cancel_url);
 
-
                 $api_key    = get_option('blockonomics_api_key');
+
                 if ($api_key == '') {
-                    if (version_compare($woocommerce->version, '2.1', '>=')) {
-                        wc_add_notice(__('Sorry, but there was an error processing your order. Please try again or try a different payment method. (plugin not configured)', 'blockonomics-bitcoin-payments'), 'error');
-                    } else {
-                        $woocommerce->add_error(__('Sorry, but there was an error processing your order. Please try again or try a different payment method. (plugin not configured)', 'blockonomics-bitcoin-payments'));
-                    }
+                    $error_str = 'API Key not set. Please login to Admin and go to Blockonomics module configuration to set you API Key.';
+                    $this->displayError($error_str, $woocommerce);
                     return;
                 }
 
-                try {
-                    $blockonomics = new Blockonomics;
-                    $address = $blockonomics->new_address(get_option('blockonomics_api_key'), get_option("blockonomics_callback_secret"));
-                    $price = $blockonomics->get_price(get_woocommerce_currency());
-                } catch (Exception $e) {
-                    $address = '';
+                $blockonomics = new Blockonomics;
+                $responseObj = $blockonomics->new_address(get_option('blockonomics_api_key'), get_option("blockonomics_callback_secret"));
+                $price = $blockonomics->get_price(get_woocommerce_currency());
+
```

Comment:
Would be good if the following error code is refactored into a new method, for cleaner code